docs/robot.md

```diff
@@ -0,0 +1,136 @@
+
+_This means that by default, your **Hostname** needs to be the same of the **name of the server in Robot**_. If this does not match, we can not properly match the two entities. Once we have made this connection, we save the Robot Server Number to the field `spec.providerId` on the Node, and use this identifier for any further processing.
+
+If you absolutely need to use different names in Robot & Hostname, you can also configure the Provider ID yourself. This can be done on the `kubelet` through the flag [`--provider-id`](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/). You need to follow the format `hrobot://$SERVER_NUMBER` when setting this. If this format is not followed exactly we can not process this node.
+
+## Config Options
+
+### Credentials
+
+You need to add your Robot credentials into the secret `hcloud`:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hcloud
+  namespace: kube-system
+type: Opaque
+stringData:
+  token: your-hcloud-token
+  robot-user: your-robot-user-name
+  robot-password: your-robot-password
+```
+
+If you only plan to use a single Robot server, you can also use an "Admin login" for this server instead of the account credentials.
+
+Then you can enable the Robot Support through the environment variable `ROBOT_ENABLE=true` or the Helm Chart value `robot.enabled: true`.
+
+You will also need to [disable Network support](#routes--private-networks) through the Helm Chart value `network.enabled: false`. If you use plain Kubernetes manifests, make sure you use the `ccm.yaml` and not the `ccm-network.yaml`.
+
+
+## Migrating from [`syself/hetzner-cloud-controller-manager`](https://github.com/syself/hetzner-cloud-controller-manager)
+
+If you have previously used the Hetzner Cloud Controller Manager by Syself, you can migrate to hcloud-cloud-controller-manager. We have tried to keep the configuration & features mostly the same and backwards compatible, but there are some changes you need to be aware of.
+
+### Configuration
+
+#### Secret Name
+
+The secret is called `hcloud` in hcloud-cloud-controller-manager, while it was called `hetzner` before. Make sure to create the new secret before migrating your deployment.
+
+#### Enable Robot Support
+
+It is now required to explicitly enable support for Robot features. This is done by setting the environment variable `ROBOT_ENABLE=true` on the container, or by setting the value `robot.enabled: true` in the Helm Chart.
+
+### Feature & behaviour changes
+
+#### Provider ID
+
+The format of the Provider ID changed from `hcloud://bm-$SERVER_NUMBER` to `hrobot://$SERVER_NUMBER`. For compatibility, we still read from the `hcloud://bm-` prefix, but any new nodes will have the `hrobot://` prefix.
+
+If you read from this value, you should amend your parsing for the new format.
+
+#### Load Balancer Targets
+
+We changed the behaviour to now remove any Robot Server targets from the Load Balancer, even if Robot support is not enabled.
```

Comment:
```suggestion
We remove any Robot Server targets from the Load Balancer, even if Robot support is not enabled.
```

I don't really understand this. What are the condition to remove the server target ?

Reply:
- Now: If HCCM does not know about the target -> remove it.
- In HCCM previously: If the target is a cloud server & HCCM does not know about it -> remove it.
- In syself fork: If Robot support is not enabled, crash when reconciling LB targets

Reply:
```suggestion
Now, any Robot Server targets are removed from the Load Balancer, even if Robot support is not enabled.
```